### PR TITLE
Blog app: Add devise

### PR DESCRIPTION
**Pull request by BB-Simon**
**The Blog app** will be a classic example of a blog website which will be a fully functional website that will show the list of posts and empower readers to interact with them by adding comments and liking posts.

### In this PR:
- [x] Remove current_user method in ApplicationController, devise will provide us with one.
- [x] Install and setup devise.
      - Can register a new user.
      - User logs in with a combination of email and password.
      - Hashed password should be stored in the database.
      - Ask for confirmation of email.
      - Can reset password.
- [x] Modify the views in devise for registration and login to match the wireframes from the Sneak Peek and your styling.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (danmainah) [inline app/models/user.rb]: - [x] @BB-Simon and @MNisarAli Great work setting up the devisee for authentication, however, you forgot to add `:confirmable` feature that allows  application to asks for confirmation of email after they register.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @BB-Simon and @MNisarAli , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Ensured the users can register and login successfully using devise.
✔️ The login and registration page design matches the wireframes in the [sneak peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md) of the project.wireframes in the [sneak peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md) of the project.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/models/user.rb (hunk 1 above)
Comment (BB-Simon) [reply app/models/user.rb]: Hi @danmainah 
Thanks for your review, I have done the required.

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @BB-Simon, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] Good UI design :white_check_mark: 
- [x] Application uses `devise` gem for authentication of users :white_check_mark: 
- [x] Pull request has a descriptive title and summary :white_check_mark: 
- [x] Implemented all the changes from the previous reviewer :white_check_mark:   

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by BB-Simon** [comment]
Hi @Felix45 
Thanks for approval:)

**Pull request merged**